### DISCUSSION TITLE
[非Bug]:ucf_drag_ModR.uc.js相关问题

### DISCUSSION THREAD

**Discussion by laulguo** [benzBrake/FirefoxCustomize#19] [Q&A]
### 你在用什么浏览器

Firefox

### 浏览器版本号

108.0.1

### 你在用什么 UC 环境(引导器)

其他

### 哪个脚本有问题

ucf_drag_ModR.uc.js

### 脚本有什么问题？

请问一下，我想要修改自带的U: /D:这样的为任意方向，都触发相同动作，应该怎样修改呢

### 错误日志

```shell
*
```

**Comment by benzBrake** [comment]
如果你是想要斜线也生效就得手势逻辑
如果只是想 UUD URD UDD ULD 对应同一个的动作，你写4个配置不就行了

**Comment by laulguo** [comment]
> 如果你是想要斜线也生效就得手势逻辑 如果只是想 UUD URD UDD ULD 对应同一个的动作，你写4个配置不就行了

"    dragover(e) {
        var { x, y } = this, cx = e.screenX, cy = e.screenY;
        var dx = cx - x, ax = Math.abs(dx), dy = cy - y, ay = Math.abs(dy);
        if (ax < 10 && ay < 10) return;

        this.x = cx; this.y = cy;
        var dir = ax > ay ? dx > 0 ? "R" : "L" : dy > 0 ? "D" : "U";
        if (this.dir.endsWith(dir)) return;

        dir = this.dir += dir;
        var obj;
        if (e.shiftKey) {
            obj = this.type[dir + "-Shift"];
        } else {
            obj = this.type[dir];
        }
"
这一段是辨别移动方向的吧，我可以不可以把他改成只要dxdy任意一个不等于0就可以呢？主要是我不太知道这个怎么写

**Comment by laulguo** [comment]
> 如果你是想要斜线也生效就得手势逻辑 如果只是想 UUD URD UDD ULD 对应同一个的动作，你写4个配置不就行了

另外这个有多段手势判定，哪一段可以把多段手势判定去改掉吗？

**Comment by laulguo** [comment]
`
//        var txt = `${obj ? "鼠标" : "未知"
//           }手势: ${dir + (obj ? "  " + obj.name : "")}`;
`
注释掉手势名
`
        var dir = ax > ay ? dx > 0 ? "U" : "U" : dy > 0 ? "U" : "U";
`
全部改成了U,解决了问题。。
